src/metaxy/metadata_store/delta.py

```diff
@@ -0,0 +1,447 @@
+        store = self._get_object_store()
+        prefix = self._feature_relative_path(feature_key).rstrip("/")
+        if prefix:
+            prefix = f"{prefix}/"
+        stream = store.list(prefix=prefix)
+        for chunk in stream:
+            paths = [obj["path"] for obj in chunk]
+            if paths:
+                store.delete(paths)
+
+    def _ensure_remote_root_accessible(self) -> None:
+        if not self._is_remote:
+            return
+
+        store = self._get_object_store()
+        from obstore.exceptions import BaseError  # type: ignore[import-not-found]
+        try:
+            store.list_with_delimiter()
+        except BaseError as exc:  # pragma: no cover - depends on backend failures
+            raise RuntimeError(
+                f"Unable to access remote DeltaMetadataStore root {self._root_uri}: {exc}"
+            ) from exc
+
+    def _get_object_store(self) -> ObjectStore:
+        """Instantiate (lazily) an obstore client for remote roots."""
+        if not self._is_remote:
+            msg = "Object store access requested for a non-remote DeltaMetadataStore"
+            raise RuntimeError(msg)
+        if self._object_store is None:
+            try:
+                from obstore import store as obstore_store  # type: ignore[import-not-found]
+            except ImportError as exc:  # pragma: no cover - handled at runtime
+                raise RuntimeError(
+                    "obstore is required for remote DeltaMetadataStore paths. "
+                    "Install metaxy[delta] to include the dependency."
+                ) from exc
+
+            self._object_store = obstore_store.from_url(
+                self._root_uri,
+                **self._object_store_kwargs,
+            )
+
+        return self._object_store
+
+    def read_metadata_in_store(
+        self,
+        feature: FeatureKey | type[BaseFeature],
+        *,
+        feature_version: str | None = None,
+        filters: Sequence[nw.Expr] | None = None,
+        columns: Sequence[str] | None = None,
+    ) -> nw.LazyFrame[Any] | None:
+        """Read metadata stored in Delta for a single feature."""
+        import deltalake  # pyright: ignore[reportMissingImports]
+
+        self._check_open()
+
+        feature_key = self._resolve_feature_key(feature)
+        table_uri = self._feature_uri(feature_key)
+        if not self._table_exists(table_uri):
+            return None
+
+        delta_table = deltalake.DeltaTable(
+            table_uri,
+            storage_options=self._storage_options_payload(),
+        )
+
+        # Use column projection for efficiency if columns are specified
+        if columns is not None:
+            # Need to ensure system columns are included for filtering
+            cols_to_read = set(columns)
+            if feature_version is not None:
+                cols_to_read.add("metaxy_feature_version")
+            arrow_table = delta_table.to_pyarrow_table(columns=list(cols_to_read))
```

Comment:
This means all the reads are going to be eager. 
We have to use `pl.scan_delta` instead to enable lazy reads.